src/redshift-mcp-server/awslabs/redshift_mcp_server/consts.py

```diff
@@ -0,0 +1,136 @@
+# limitations under the License.
+
+"""Redshift MCP Server constants."""
+
+# Defaults
+DEFAULT_AWS_REGION = 'us-east-1'
+DEFAULT_LOG_LEVEL = 'WARNING'
+
+# Timeouts (seconds), etc
+CLIENT_TIMEOUT = 60
+DATA_CLIENT_TIMEOUT = 60
+QUERY_TIMEOUT = 3600
+QUERY_POLL_INTERVAL = 2
+
+# Best practices
+
+CLIENT_BEST_PRACTICES = """
+## AWS Client Best Practices
+
+### Authentication
+
+- Default AWS credentials chain (IAM roles, ~/.aws/credentials, etc.).
+- AWS_PROFILE environment variable (if set).
+- AWS_REGION environment variable (if set).
+"""
+
+REDSHIFT_BEST_PRACTICES = """
+## Amazon Redshift Best Practices
+
+### Query Guidelines
+
+- Always specify the database and schema when referencing objects to avoid ambiguity.
+- Leverage distribution in WHERE and JOIN predicates and sort keys in ORDER BY for optimal query performance.
+- Use LIMIT clauses for exploratory queries to avoid large result sets.
+- Analyze table to update table statistics if it is not updated or too off before making a decision on the query structure.
+- Prefer explicitly specifying columns in SELECT over "*" for better performance.
+
+### Connection Guidelines
+
+- We are use the Redshift API and Redshift Data API.
+- Leverage IAM authentication when possible instead of secrets (database passwords).
+"""
+
+# SQL queries
+
+SVV_REDSHIFT_DATABASES_QUERY = """
+SELECT
+    database_name,
+    database_owner,
+    database_type,
+    database_acl,
+    database_options,
+    database_isolation_level
+FROM pg_catalog.svv_redshift_databases
+ORDER BY database_name;
+"""
+
+SVV_ALL_SCHEMAS_QUERY = """
+SELECT
+    database_name,
+    schema_name,
+    schema_owner,
+    schema_type,
+    schema_acl,
+    source_database,
+    schema_option
+FROM pg_catalog.svv_all_schemas
+WHERE database_name = {}
+ORDER BY schema_name;
+"""
+
+SVV_ALL_TABLES_QUERY = """
+SELECT
+    database_name,
+    schema_name,
+    table_name,
+    table_acl,
+    table_type,
+    remarks
+FROM pg_catalog.svv_all_tables
+WHERE database_name = {} AND schema_name = {}
+ORDER BY table_name;
+"""
+
+SVV_ALL_COLUMNS_QUERY = """
+SELECT
+    database_name,
+    schema_name,
+    table_name,
+    column_name,
+    ordinal_position,
+    column_default,
+    is_nullable,
+    data_type,
+    character_maximum_length,
+    numeric_precision,
+    numeric_scale,
+    remarks
+FROM pg_catalog.svv_all_columns
+WHERE database_name = {} AND schema_name = {} AND table_name = {}
+ORDER BY ordinal_position;
+"""
+
+# SQL guardrails
+
+# Single-lines comments.
+re_slc = r'--.*?$'
+
+
+def re_mlc(g: int) -> str:
+    """Multi-line comments, considering balanced recursion."""
+    return rf'(?P<mlc{g}>(?:\/\*)(?:[^\/\*]|\/[^\*]|\*[^\/]|(?P>mlc{g}))*(?:\*\/))'
+
+
+def re_sp(g: int) -> str:
+    """Whitespaces, comments, semicolons which can occur between words."""
+    return rf'({re_slc}|{re_mlc(g)}|\s|;)'
+
+
+# We consider `(END|COMMIT|ROLLBACK|ABORT) [WORK|TRANSACTION]` as a breaker for the `BEGIN READ ONLY; {sql}; END;`
```

Comment:
Is `CREATE` something to include?

Reply:
No. CREATE is not a transaction breaker. The idea I implemented is to use a read-only transaction and make sure it is not broken by the breaker commands as I have shown. Tracking all the possible DDL/DML variants is not reliable.